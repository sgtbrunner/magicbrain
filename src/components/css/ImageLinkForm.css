.custom-shadow-w {
	width: 700px;
	box-shadow:0px 0px 4px 4px rgba(0, 0, 0, 0.3);
}