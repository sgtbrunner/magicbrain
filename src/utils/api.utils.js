const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
