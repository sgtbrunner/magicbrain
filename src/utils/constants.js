export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
