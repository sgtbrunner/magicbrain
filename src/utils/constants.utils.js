const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
