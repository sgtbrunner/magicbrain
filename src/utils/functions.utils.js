export const getUserFromLocalStorage = () => localStorage.getItem('user');
