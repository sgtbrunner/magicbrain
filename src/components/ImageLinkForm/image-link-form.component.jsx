import React from 'react';
import PropTypes from 'prop-types';

import './image-link-form.styles.css';

const ImageLinkForm = ({ onInputChange, onButtonClick }) => {
  return (
    <div>
      <p className="f3 smaller">
        The <b>MagicBrain</b> will detect faces in your pictures. Give it a try!
      </p>
      <div className="centered smaller">
        <div className="pa4 br3 custom-shadow-w form centered smaller">
          <input
            className="f4 pa2 w-70 centered ba"
            type="text"
            placeholder="enter the image URL here"
            onChange={onInputChange}
          />
          <button
            className="w-30 f4 link ph3 pv2 dib light bg-light-purple noselect"
            onClick={onButtonClick}
            type="button"
          >
            Detect
          </button>
        </div>
      </div>
    </div>
  );
};

ImageLinkForm.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  onButtonClick: PropTypes.func.isRequired,
};

export default ImageLinkForm;
