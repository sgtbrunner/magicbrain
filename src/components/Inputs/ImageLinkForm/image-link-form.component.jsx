import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './image-link-form.styles.css';

const ImageLinkForm = ({ onFormSubmit }) => {
  const [imageInput, setImageInput] = useState('');

  const onInputChange = (event) => {
    setImageInput(event.target.value);
  };

  const onInputSubmit = (event) => {
    event.preventDefault();
    onFormSubmit(imageInput);
  };

  return (
    <div>
      <p className="f3 smaller">
        The <b>MagicBrain</b> will detect faces in your pictures. Give it a try!
      </p>
      <div className="centered smaller">
        <form className="pa4 br3 custom-shadow-w form centered smaller">
          <input
            className="f4 pa2 w-70 centered ba"
            type="text"
            placeholder="enter the image URL here"
            onChange={onInputChange}
          />
          <button
            className="w-30 f4 link ph3 pv2 dib light bg-light-purple noselect"
            onClick={onInputSubmit}
            type="submit"
            disabled={!imageInput}
          >
            Detect
          </button>
        </form>
      </div>
    </div>
  );
};

ImageLinkForm.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
};

export default ImageLinkForm;
