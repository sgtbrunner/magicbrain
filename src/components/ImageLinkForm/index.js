import ImageLinkForm from './image-link-form.component';

export default ImageLinkForm;
