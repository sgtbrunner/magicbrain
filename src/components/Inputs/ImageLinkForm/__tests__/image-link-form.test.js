import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { mountComponent } from '../../../../utils/test.utils';
import ImageLinkForm from '..';

const BUTTON = 'button';

const mountImageLinkFormComponent = () =>
  mountComponent({ component: ImageLinkForm, props: { onFormSubmit: jest.fn() } });

describe('ImageLinkForm component', () => {
  it('should match snapshot', () => {
    const { container } = mountImageLinkFormComponent();

    expect(container).toMatchSnapshot();
  });

  it(`should render a button with 'Detect' label`, () => {
    mountImageLinkFormComponent();

    const button = screen.getByRole(BUTTON);
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('Detect');
  });

  it('should have button initially disabled', () => {
    mountImageLinkFormComponent();

    const button = screen.getByRole(BUTTON);
    expect(button).toBeDisabled();
  });

  it('should enable button if user fills in the input field', () => {
    mountImageLinkFormComponent();

    const input = screen.getByRole('textbox');

    userEvent.clear(input);
    userEvent.type(input, 'http:my-domain.com/my-image.jpg');

    const button = screen.getByRole(BUTTON);
    expect(button).toBeEnabled();
  });
});
