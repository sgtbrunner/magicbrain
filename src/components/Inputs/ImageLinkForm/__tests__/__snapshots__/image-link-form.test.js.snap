// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLinkForm component should match snapshot 1`] = `
<div>
  <label
    class="f3"
    for="image-link-form"
  >
    The 
    <b>
      MagicBrain
    </b>
     will detect faces in your pictures. Give it a try!
    <form
      class="pa4 br3 mt3 custom-shadow-w form smaller center"
      data-testid="image-link-form"
      id="image-link-form"
    >
      <input
        class="f4 pa2 w-70 ba"
        placeholder="enter the image URL here"
        type="text"
      />
      <button
        class="w-30 f4 link ph3 pv2 dib light bg-light-purple noselect"
        disabled=""
        type="submit"
      >
        Detect
      </button>
    </form>
  </label>
</div>
`;
